### DISCUSSION TITLE
"Normal" Resource Usage (CPU/RAM)? Proxmox LXC

### DISCUSSION THREAD

**Discussion by tokenwizard** [mediacms-io/mediacms#1062] [Q&A]
**Describe the issue**
When Uploading/Transcoding media, CPU and RAM usage it taking everything I give it and want more. When trying to upload multiple files at once, they eventually all fail to upload. 

**To Reproduce**
Steps to reproduce the issue:
1. Run MediaCMS (Docker)
2. Log in as Admin
3. Upload a Media file
4. Watch CPU/RAM usage

**Expected behavior**
The media gets uploaded and transcoded and is available to view, without grinding the entire system to a halt. 

**Screenshots**
![image](https://github.com/user-attachments/assets/ff19e952-ec1e-44ee-9aa8-fd867dad8aa7)

**Environment (please complete the following information):**
 - OS: Arch Linux (Proxmox LXC)
 - Installation method: Docker
 - Firefox 130.0 / Zen Browser 1.0.1-a.2

**Additional context**
I run many apps via Docker in LXC containers, including Plex and Frigate NVR, and even my Frigate install that is running Object Detection on 10 HD Cameras via an NVIDIA GPU (All in Docker) and it doesn't use half the CPU/RAM allotment as this does just trying to upload one 1080p video, so I feel like I am missing something critical. Is there a way to make sure the GPU is used?

**Comment by tokenwizard** [comment]
Update: After several minutes (file upload completed about 10min ago) I am reading even higher resource usage and at this point SSH to the LXC container is failing and the MediaCMS service seems to have stopped. I refreshed the browser and not the service no longer seems to be running. I will eave it to see if maybe it completes the transcoding and calms down before restarting the container.

**Reply by bigbeka** [reply]
@tokenwizard were you able to resolve this issue? If so how? I have been having this issue for many months now, the only way is to wait with frozen instance of MCMS or throw more RAM at it until the transcribing finishes.

<img width="845" height="339" alt="Screenshot 2025-10-21 at 10 01 02" src="https://github.com/user-attachments/assets/f9125d05-60a1-4e48-838c-3003aca9d53b" />

**Comment by mgogoulos** [comment]
Hi @tokenwizard , I'll move this to the discussions as I'm not sure it's an issue. I personally haven't used this environment and am not familiar with Proxmox, so cannot say how it could be related to create the extra overhead or not...

**Comment by tokenwizard** [comment]
Ok, thanks. To clarify Proxmox is just a Hypervisor OS that allows for running VMs and LXD/LXC containers and assigning specific CPU/RAM Allocations to each one. So the usages shown above are specific to the LXC that is running MediaCMS and nothing else. That would not include any overhead form the baremetal Proxmox server.

**Comment by tokenwizard** [comment]
I definitely seems to be the transcoding process that spikes the CPU/RAM usage. I just upload a single video, as soo as the upload completes, I hear the server fans start whirring more loudly and the CPU and RAM both spike to nearly 100%, even with 6 vCPUs and 12GB RAM. 

So I'm thinking if I can get the transcode process to use my GPU, then it may help. Not sure if that is something that can be configured in MediaCMS or not.

**Comment by mgogoulos** [comment]
I think you have to change ffmpeg settings so that it works on a GPU, that's the only place that processing power is needed. Checkout through search what have been writen related to GPU

**Reply by tokenwizard** [reply]
Ok, I will take a look at the Issues for GPU related notes. ffmpeg is built into the docker container, right? So likely this would be a change that would need to be made in the docker container and would be lost on pulling an update.

**Reply by tokenwizard** [reply]
This doesn't look too promising, lol.
![image](https://github.com/user-attachments/assets/09820ed5-5173-4778-a9c0-76449af25bbc)

Seems like only one other discussion of GPU and it was closed. So outside of installing bare-metal I may need to recompile ffmpeg in the container itself and then commit that to a new container.

**Reply by tokenwizard** [reply]
It looks like all three of these services/containers include ffmpeg (I am able to run it within each of these containers). 

Do you know if all of these used in the transcode process? 

![image](https://github.com/user-attachments/assets/42f58a8f-1dd8-4cde-ac81-a68153d467b7)

**Reply by mgogoulos** [reply]
Hi, the Docker image is the same, but only celery_worker is loading code that uses ffmpeg

**Comment by mgogoulos** [comment]
There are some previous discussions too (besides issues)
https://github.com/mediacms-io/mediacms/discussions/314
https://github.com/mediacms-io/mediacms/discussions/862

maybe there's something there that gives hints...
